Sources/NIOFileSystem/Internal/Concurrency Primitives/TokenBucket.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Type modeled after a "token bucket" pattern, which is similar to a semaphore, but is built with
+/// Swift Concurrency primitives.
+///
+/// This is an adaptation of the TokenBucket found in Swift Package Manager.
+/// Instead of using an ``actor``, we define a class and limit access through
+/// ``NIOLock``.
+
+import DequeModule
+import NIOConcurrencyHelpers
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+class TokenBucket {
```

Comment:
Reviewing this myself now, it looks like I added the imports between the `class` definition and its comment. Surprised that no linter is barfing at me here.